ci: add release build script

a110605 opened the pull request:
<!-- This template is for Devs to give QA details before moving the issue To-Test -->
### Summary
Add release build script, build ui-extension source code and push tar.gz to GCP cloud space
- https://releases.rancher.com/harvester-ui/dashboard/latest.tar.gz
- https://releases.rancher.com/harvester-ui/plugin/harvester-latest.tar.gz

Below artifacts will get updated when main branch has any PR merged
- https://releases.rancher.com/harvester-ui/dashboard/latest/index.html
- https://releases.rancher.com/harvester-ui/plugin/harvester-latest/harvester-latest.umd.min.js


TODO:
- merge this PR and check harvester ISO build standalone UI work as usual
- disable master branch build artifact from harvester/dashboard (PR: https://github.com/harvester/dashboard/pull/1250)
- add v1.5.0 feature flag and create `harvester-release-v1.5` branch in harvester-ui-extension repo.

### PR Checklists
- Do we need to backport this PR change to the [Harvester Dashboard](https://github.com/harvester/dashboard)?
    - [ ] Yes, the relevant PR is at:
- Are backend engineers aware of UI changes?
    - [ ] Yes, the backend owner is:

### Related Issue #
https://github.com/harvester/harvester/issues/7089

### Test screenshot/video
<!-- Attach screenshot or video of the changes and eventual comparison if you find it necessary -->

### Extra technical notes summary
<!-- Outline technical changes which may pass unobserved or may help to understand the process of solving the issue -->

a110605 commented on .github/workflows/build-and-upload.yaml:
```diff
@@ -0,0 +1,205 @@
+  #         tags: rancher/harvester-ui:${{ github.ref_name }}
+
+  build-and-upload-hosted:
+    name: Build & Upload Hosted
```
Not sure we still needed build host, from harvester/dashboard [README.md](https://github.com/harvester/dashboard#other-building-modes), this step is build `ui-index` file and push to GCP. e.g.
```
https://releases.rancher.com/harvester-ui/dashboard/latest/index.html
```

torchiaf replied:
I think we need it for the standalone mode

torchiaf commented on .github/workflows/build-and-upload.yaml:
```diff
@@ -0,0 +1,205 @@
+        # https://github.com/google-github-actions/upload-cloud-storage
+        with:
+          path: ${{steps.build-hosted.outputs.BUILD_HOSTED_DIR}}
+          destination: releases.rancher.com/harvester-ui/dashboard/${{ steps.build-hosted.outputs.BUILD_HOSTED_LOCATION }}
```
Could you please replace the `destination` name using a mock before merging? Something like `test-release`.
This way we can push a test artifacts without overwrite the current master `branch` release and we will easily identify test artifacts to delete from cloud storage.

torchiaf commented on .github/workflows/build-and-upload.yaml:
```diff
@@ -0,0 +1,205 @@
+        uses: 'google-github-actions/upload-cloud-storage@v2'
+        with:
+          path: ${{steps.build-embedded.outputs.BUILD_EMBEDED_TGZ}}
+          destination: releases.rancher.com/harvester-ui/${{ env.REPO }}
```
Change `destination` name , see comment above.

torchiaf commented on .github/workflows/build-and-upload.yaml:
```diff
@@ -0,0 +1,205 @@
+            cache-control: no-cache,must-revalidate
+          process_gcloudignore: false
+
+  build-and-upload-harvester-plugin:
```
We can remove this Job.
- Rancher version 2.9 and lower is not compatible with the extension for Vue compatibility
- Rancher 2.10 and higher will use the extension installation

a110605 replied:
Let's keep this job since harvester [dockerFile](https://github.com/harvester/harvester/blob/9609b9038f3724e584cfc432aaedd0274b6d8cc8/package/Dockerfile#L39) still pull below tar.gz

- https://releases.rancher.com/harvester-ui/plugin/harvester-latesttar.gz
- https://releases.rancher.com/harvester-ui/plugin/harvester-release-harvester-v1.5.tar.gz

torchiaf reviewed (changes requested):
I left some comments:

- `build-and-upload-harvester-plugin:` to be removed
- let's change the installation urls

After this, we could go ahead this way:

- Merge changes
- Push some test releases
- If the new artifacts looks good, replace the mock urls with the real ones
- Open a GH issue to the eio team to remove tests artifacts.

What do you think @a110605 ?

a110605 commented:
> I left some comments:
> 
> * `build-and-upload-harvester-plugin` job to be removed
> * let's change the installation urls
> 
> After this, we could go ahead this way:
> 
> * Merge changes
> * Push some test releases
> * If the new artifacts looks good, replace the mock urls with the real ones
> * Open a GH issue to the eio team to remove tests artifacts.
> 
> What do you think @a110605 ?

Agreed. I updated all upload folder name to `test-release` and modified a [mock word](https://github.com/harvester/harvester-ui-extension/pull/56/commits/bad78f00bb757cb62332a9d11ae99bf018a76b3b) in setting page. 
We can merge this PR and push a new commit in main branch. Then use below `ui-index` to test in standalone mode 
```
https://releases.rancher.com/harvester-ui/dashboard/test-release/index.html
```

a110605 commented:
Done testing build host artifact is good for standalone mode.

<img width="1494" alt="Screenshot 2025-01-06 at 5 08 55 PM" src="https://github.com/user-attachments/assets/03b8110a-bf2a-4c5c-9eeb-a7a00ca84d73" />
